File: 2037-count-square-sum-triples/count-square-sum-triples.py
class Solution:
    def countTriples(self, n: int) -> int:
        squares = {x * x for x in range(1, n + 1)}
        count = 0
        for i in range(1, n + 1):
            for j in range(1, n + 1):
                if i * i + j * j in squares:
                    count += 1
        return count

File: 2037-count-square-sum-triples/count-square-sum-triples.java
import java.util.*;

class Solution {
    public int countTriples(int n) {
        HashSet<Integer> squares = new HashSet<>();
        for (int x = 1; x <= n; x++) {
            squares.add(x * x);
        }
        int count = 0;
        for (int i = 1; i <= n; i++) {
            int i2 = i * i;
            for (int j = 1; j <= n; j++) {
                int s = i2 + j * j;
                if (squares.contains(s)) {
                    count++;
                }
            }
        }
        return count;
    }
}
